src/main/java/com/google/devtools/build/lib/cmdline/RepositoryName.java

```diff
@@ -139,13 +140,22 @@ public static Pair<RepositoryName, PathFragment> fromPathFragment(
    */
   @Nullable private final RepositoryName ownerRepoIfNotVisible;
 
-  private RepositoryName(String name, @Nullable RepositoryName ownerRepoIfNotVisible) {
+  /**
+   * If ownerRepoIfNotVisible is not null, this field stores the suffix to be appended to the error
+   */
+  @Nullable private final String didYouMeanSuffix;
```

Comment:
I could offer to refactor this class into a sealed base class with a single field and a subclass with all three fields. The memory savings wouldn't be that significant (8 bytes per instance, of which there are most likely less than 100,000 even for very large repos).